scipy/optimize/_minimize.py

```diff
@@ -109,34 +109,37 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
         obey any specified `bounds`.
     hess : {callable, '2-point', '3-point', 'cs', HessianUpdateStrategy}, optional
         Method for computing the Hessian matrix. Only for Newton-CG, dogleg,
-        trust-ncg, trust-krylov, trust-exact and trust-constr. If it is
-        callable, it should return the Hessian matrix:
+        trust-ncg, trust-krylov, trust-exact and trust-constr.
+        If it is callable, it should return the Hessian matrix:
 
             ``hess(x, *args) -> {LinearOperator, spmatrix, array}, (n, n)``
 
         where ``x`` is a (n,) ndarray and ``args`` is a tuple with the fixed
         parameters. LinearOperator and sparse matrix returns are only allowed
-        for 'trust-constr' method. Alternatively (not available for Newton-CG
-        or dogleg), the keywords {'2-point', '3-point', 'cs'} select a finite
-        difference scheme for numerical estimation. Or, objects implementing
-        the `HessianUpdateStrategy` interface can be used to approximate the
-        Hessian. Available quasi-Newton methods implementing this interface
-        are:
+        for the trust-constr method.
+        dogleg and trust-exact can only be used with a callable returning
+        an (n, n) array.
+        It is not mandatory to supply this keyword for trust-constr or
+        Newton-CG.
+        For the Newton-CG, trust-krylov, trust-ncg, and trust-constr methods
+        the keywords {'2-point', '3-point', 'cs'} can be used to select a
+        finite difference scheme for numerical estimation. Or, objects
+        implementing the `HessianUpdateStrategy` interface can be
```

Comment:
Starting from here I am lost.

Reply:
A table may make more sense. I might write some more a test function to check what's written in the table.

Reply:
Thanks for doing the table, it looks great and really clarifies things! I would add a note in here to direct the user the the note section for a table. Also we could have a smaller text here and move the long explanation along with the table. Your call.

Reply:
I believe I've already done that in the PR:

> Not all of the options are available for each of the methods; for availability refer to the notes.

Reply:
Indeed, sorry GH was showing me the old one since we have the discussion on an outdated code... Then all good. Do you still want to add tests or is this ready for you?

Reply:
I'm happy with the PR being merged as-is, unless others have comments.

Reply:
Great, then let's get this in. We can always update things later and right now it's a clear improvement.